### PR TITLE
Fixed PHP's array_merge_recursive issue

### PR DESCRIPTION
**Pull request by hinablue**
Hello!

Type: bug fix 
Link to issue:

In raising this pull request, I confirm the following (please check boxes):
- [x] I have read and understood the Contributing Guidelines?
- [x] I have checked that another pull request for this purpose does not exist.
- [ ] I wrote some tests for this PR.
  Small description of change:

RESEND PR: https://github.com/phalcon/cphalcon/pull/12307

There (maybe) is the PHP core issue, not Phalcon. And I rewrite the array_merge_recursive like here,
[http://php.net/manual/en/function.array-merge-recursive.php#96201](http://php.net/manual/en/function.array-merge-recursive.php#96201)

Use `Phalcon\Config\Adapter\Ini`, If I have ini file like,

```
[step]
step1.1 = A
step1.2 = B
step1.3 = C
```

Output will be,

```
array(1) {
  ["step"]=>
  array(1) {
    ["step1"]=>
    array(2) {
      [1]=>
      string(1) "B"
      [2]=>
      string(1) "C"
    }
  }
}
```

The 'A' will missing, but If,

```
[step]
step1.5 = X
step1.1 = A
step1.2 = B
step1.3 = C
```

Output will be,

```
array(1) {
  ["step"]=>
  array(1) {
    ["step1"]=>
    array(2) {
      [5]=>
      string(1) "X"
      [1]=>
      string(1) "A"
      [2]=>
      string(1) "B"
      [3]=>
      string(1) "C"
    }
  }
}
```

This issue is PHP's array_merge_recursive (maybe), so I rewrite an new function to do this in `ini.zep` and output correct what we want.

There is ini file,

```
[qq]
step1.1 = A
step1.2 = B
step1.3 = G
step2.1 = A
step2.2 = B
step2.3 = C
step12[3] = AA
step12[4] = BB
step12[5] = CC
step5.3 = AAA
step5.4 = BBBB
step5.1 = GGGG
step5.2 = TTTT
```

And output will be,

```
object(Phalcon\Config\Adapter\Ini)#1 (1) {
  ["qq"]=>
  object(Phalcon\Config)#2 (4) {
    ["step1"]=>
    object(Phalcon\Config)#3 (3) {
      ["1"]=>
      string(1) "A"
      ["2"]=>
      string(1) "B"
      ["3"]=>
      string(1) "G"
    }
    ["step2"]=>
    object(Phalcon\Config)#4 (3) {
      ["1"]=>
      string(1) "A"
      ["2"]=>
      string(1) "B"
      ["3"]=>
      string(1) "C"
    }
    ["step12"]=>
    object(Phalcon\Config)#5 (3) {
      ["3"]=>
      string(2) "AA"
      ["4"]=>
      string(2) "BB"
      ["5"]=>
      string(2) "CC"
    }
    ["step5"]=>
    object(Phalcon\Config)#6 (4) {
      ["3"]=>
      string(3) "AAA"
      ["4"]=>
      string(4) "BBBB"
      ["1"]=>
      string(4) "GGGG"
      ["2"]=>
      string(4) "TTTT"
    }
  }
}
```

Thanks

### CODE REVIEW

File: phalcon/config/adapter/ini.zep [hunk 1]
```diff
@@ -68,113 +68,161 @@ use Phalcon\Config\Exception;
...
-				if preg_match("/[.]+/", ini) {
-					return (double) ini;
-				} else {
-					return (int) ini;
-				}
-			}
-		}
-		return ini;
-	}
+    /**
```
Comment (sergeyklay) [inline phalcon/config/adapter/ini.zep]: Tabs must be used to indent lines; spaces are not allowed

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Comment by sergeyklay** [comment]
Could you please open issue and refer it here. Also fix please indents and check checkboxes in PR description ^

**Review by sergeyklay — APPROVED** [review]

File: phalcon/config/adapter/ini.zep [hunk 2]
```diff
@@ -93,10 +93,10 @@ class Ini extends Config
 			if typeof directives == "array" {
 				let sections = [];
 				for path, lastValue in directives {
-					let sections[] = this->_parseIniString((string)path, lastValue);
+					let sections[] = this->_parseIniString((string) path, lastValue);
 				}
 				if count(sections) {
-					let config[section] = call_user_func_array("array_merge_recursive", sections);
+					let config[section] = call_user_func_array([this, "array_merge_recursive"], sections);
```
Comment (SidRoberts) [inline phalcon/config/adapter/ini.zep]: Your method is called "_array_merge_recursive_distinct".

**Review by SidRoberts — CHANGES_REQUESTED** [review]

**Comment by sergeyklay** [comment]
@hinablue Could you please rebase? Also update please CHANGELOG.md

**Comment by sergeyklay** [comment]
@hinablue  Ping

**Comment by Jurigag** [comment]
Please send commits related only to this PR. Like squash all commits. You most likely rebased wrong repo.

**Comment by sergeyklay** [comment]
@hinablue 
Could you please submit just the commits that belong to the PR? Also update please CHANGELOG.md

**Comment by hinablue** [comment]
Sorry for the remove the branch in my fork again. I am troubled by the git rebase.
Who can tell me how to do git rebase correctly?

**Comment by sergeyklay** [comment]
@hinablue

- https://www.atlassian.com/git/tutorials/merging-vs-rebasing/summary
- https://sellarafaeli.wordpress.com/2014/01/16/git-rebase-for-dummies/
- https://git-scm.com/book/en/v2/Git-Branching-Rebasing

**Pull request closed without merging**